src/main/java/com/moviecat/servicesImpl/implementations/BaseService.java

```diff
@@ -56,6 +56,8 @@ public static boolean isAlive() {
         DbProvider.connection().delete(obj);
     }
 
+    public static <T> void delete(Class<T> obj, Query<T> query, Object id){ DbProvider.connection().delete(query);}
```

Comment:
There are 2 delete methods.
1 takes the actual object.
1 takes the query. We should modify the `delete(Query<T> obj)` to `delete(Class<T> clazz, Query<T> obj)`
The Query should already contain the object id we wish to delete.
That makes your additional method redundant.

Reply:
Makes sense. Will handle this.